pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java

```diff
@@ -103,4 +100,60 @@ public String processQueryPost(String query) {
       throw new WebApplicationException(e, Response.Status.INTERNAL_SERVER_ERROR);
     }
   }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("sql")
+  @ApiOperation(value = "Querying pinot in sql")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Query response"), @ApiResponse(code = 500, message = "Internal Server Error")})
+  public String processSqlQueryGet(@ApiParam(value = "Query", required = true) @QueryParam("sql") String query,
+      @ApiParam(value = "Trace enabled") @QueryParam(Request.TRACE) String traceEnabled,
+      @ApiParam(value = "Debug options") @QueryParam(Request.DEBUG_OPTIONS) String debugOptions) {
+    try {
+      ObjectNode requestJson = JsonUtils.newObjectNode();
+      requestJson.put(Request.SQL, query);
+      String queryOptions = constructSqlQueryOptions();
+      requestJson.put(Request.QUERY_OPTIONS, queryOptions);
+      if (traceEnabled != null) {
+        requestJson.put(Request.TRACE, traceEnabled);
+      }
+      if (debugOptions != null) {
+        requestJson.put(Request.DEBUG_OPTIONS, debugOptions);
+      }
+      BrokerResponse brokerResponse = requestHandler.handleRequest(requestJson, null, new RequestStatistics());
+      return brokerResponse.toJsonString();
+    } catch (Exception e) {
+      LOGGER.error("Caught exception while processing GET request", e);
+      brokerMetrics.addMeteredGlobalValue(BrokerMeter.UNCAUGHT_GET_EXCEPTIONS, 1L);
+      throw new WebApplicationException(e, Response.Status.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("sql")
+  @ApiOperation(value = "Querying pinot")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Query response"), @ApiResponse(code = 500, message = "Internal Server Error")})
+  public String processSqlQueryPost(String query) {
```

Comment:
Why not contain trace and debug_option in this endpoint?

Reply:
Everything is part of the payload in this one